orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/aws/cloudformation/DeleteCloudFormationChangeSetTask.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2019 Adevinta
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.clouddriver.tasks.providers.aws.cloudformation;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.Task;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.KatoService;
+import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import java.util.*;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class DeleteCloudFormationChangeSetTask extends AbstractCloudProviderAwareTask
+    implements Task {
+
+  @Autowired KatoService katoService;
+
+  public static final String TASK_NAME = "deleteCloudFormationChangeSet";
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    String cloudProvider = getCloudProvider(stage);
+
+    Map<String, Object> task = new HashMap<>(stage.getContext());
```

Comment:
`stage.getContext()` already returns a `Map`, why do you need to copy it?
If you do need to do the copy, consider renaming `task` to `stageData` - that's what we use pretty much everywhere else and I think it's more representative of what this is

Reply:
Good point. We made a copy of the context map because we were using the contents of the context to create the contents of the kato task operation. In that sense, both the context and task are coupled. I will make it more explicit by creating the task object from scratch and so we don't rely on assumptions.